homeassistant/components/cambridge_audio/quality_scale.yaml

```
@@ -0,0 +1,89 @@
+rules:
+  # Bronze
+  action-setup:
+    status: exempt
+    comment: |
+      This integration does not provide additional actions beyond play media which is setup by the media player entity.
+  appropriate-polling:
+    status: exempt
+    comment: |
+      This integration uses a push API. No polling required.
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions: done
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup: done
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+
+  # Silver
+  config-entry-unloading: done
+  log-when-unavailable: done
+  entity-unavailable: done
+  action-exceptions: done
+  reauthentication-flow:
+    status: exempt
+    comment: |
+      This integration does not require authentication.
+  parallel-updates:
+    status: exempt
+    comment: |
+      This integration uses a push API. No parallel poll APIs necessary.
+  test-coverage: done
+  integration-owner: done
+  docs-installation-parameters: done
+  docs-configuration-parameters:
+    status: exempt
+    comment: |
+      This integration does not have an options flow.
+  # Gold
+  entity-translations: done
+  entity-device-class: done
+  devices: done
+  entity-category: done
+  entity-disabled-by-default:
+    status: exempt
+    comment: |
+      This integration doesn't have enough / noisy entities that warrant being disabled by default.
+  discovery: done
+  stale-devices:
+    status: exempt
+    comment: |
+      This integration is not a hub and as such only represents a single device.
+  diagnostics: done
+  exception-translations: done
+  icon-translations: done
+  reconfiguration-flow: todo
+  dynamic-devices:
+    status: exempt
+    comment: |
+      This integration is not a hub and only represents a single device.
+  discovery-update-info: done
+  repair-issues:
+    status: exempt
+    comment: |
+      This integration doesn't have any cases where raising an issue is needed.
+  docs-use-cases: done
+  docs-supported-devices: done
+  docs-supported-functions: done
+  docs-data-update: done
+  docs-known-limitations: done
+  docs-troubleshooting: done
+  docs-examples: done
+
+  # Platinum
+  async-dependency: done
+  inject-websession:
+    status: exempt
+    comment: |
+      This integration uses websockets exclusively and does not make http calls.
```

This seems incorrect. WebSockets are still connections going over HTTP and can thus use a shared session.

```suggestion
  inject-websession: todo
```

---

HTTP is only used at the handshake to promote the websocket protocol, so it's not used except for the handshake. That being said, I've migrated from the websockets library to aiohttp, so I'll be able to pass in the ClientSession for that handshake process.

Suggestion accepted to reflect current status, can update in future PR.